src/Microsoft.AspNetCore.Server.Kestrel/Infrastructure/Constants.cs

```diff
@@ -20,5 +23,20 @@ internal class Constants
         /// for info on the format.
         /// </summary>
         public const string RFC1123DateFormat = "r";
+
+        private static int GetECONNRESET()
+        {
+            switch (PlatformServices.Default.Runtime.OperatingSystemPlatform)
+            {
+                case Platform.Windows:
+                    return -4077;
+                case Platform.Linux:
+                    return -104;
+                case Platform.Darwin:
+                    return -54;
+                default:
+                    throw new Exception("Unable to determine ECONNRESET value on this platform.");
```

Comment:
Do you just default it to the Linux constant? More than likely, if it hits that default block, it is some variant of Linux that isn't being detected properly.